src/pipeline.rs

```diff
@@ -439,25 +444,43 @@ fn maybe_send(r: Result<()>) {
     }
 }
 
+struct PipelineContext {
```

Comment:
What's the difference between `PipelineContext` and `PipelineAlias`, I'm a bit confused about that.

Reply:
We also have a `ConnectorContext`, which is providing contextual information and some utils to every connector and is passed in to literally all calls. I wanted to have something similar here. This context though is only used for printing the correct prefix for log messages: `[Pipeline::flow::pipeline_alias] ...`

Reply:
Perhaps to make that clearer type wise `PipelineContext` could contain `PipelineId` instead of the `String`? That way it would become obvious that The context is a higher level abstraction that could potentially grow outside of the pure ID but is just that for now.